src/js/components/CheckBox/CheckBox.js

```diff
@@ -59,6 +59,12 @@ const CheckBox = forwardRef(
   ) => {
     const theme = useContext(ThemeContext) || defaultProps.theme;
     const formContext = useContext(FormContext);
+    if (formContext.noForm !== undefined && !formContext.noForm && !name) {
+      console.warn(
```

Comment:
Over here, `undefined` is added because there is no `name` prop  added for DataTable component